Scala/Scala.sublime-syntax

```diff
@@ -77,25 +94,62 @@ contexts:
           scope: keyword.other.documentation.scaladoc.scala
         - match: '\{@link\s+[^\}]*\}'
           scope: keyword.other.documentation.scaladoc.link.scala
+
+  ascription:
+    - match: '{{typeprefix}}'
+      push: single-type-expression
+
+  lambdas:
+    - match: '(?=({{id}}|{{id}}\s*:\s*{{id}}|\([^\)]*\))\s*=>[[[:alpha:]]0-9\s\)\]\}])'
+      push:
+        - match: '{{oplookbehind}}(=>){{oplookahead}}'
+          captures:
+            1: storage.type.function.arrow.scala
+          pop: true
+        - include: lambda-declaration
+  lambda-declaration-base:
+    - match: \(
+      push:
+        - match: \)
+          pop: true
+        - include: lambda-declaration-parens
+    - match: '{{id}}'
+      scope: variable.parameter
+  lambda-declaration-parens:
+    - match: '{{typeprefix}}'
+      push:
+        - match: ','
+          pop: true
+        - match: '(?=\))'
+          pop: true
+        - include: delimited-type-expression
+    - include: lambda-declaration-base
+  lambda-declaration:
+    - match: '{{typeprefix}}'
+      push:
+        - match: '(?={{oplookbehind}}(=>){{oplookahead}})'
```

Comment:
Yep.  Lookception.  I did that just because of laziness.  I do need the lookahead, but obviously due to the lookahead, the lookbehind (and also the nested lookahead) are unnecessary.